MonumentAddons.cs

```diff
@@ -5430,6 +5467,35 @@ public virtual void HandleChanges()
                 UpdateScale();
                 UpdateBuildingGrade();
                 UpdateIOConnections();
+                UpdateSkullName();
+            }
+
+            public void UpdateSkullName() 
+            {
+                var skullName = EntityData.SkullName;
+                if (skullName == null)
+                    return;
+
+                var skullTrophy = Entity as SkullTrophy;
+                if (skullTrophy == null)
+                    return;
+
+                if (skullTrophy.inventory == null)
+                    return;
+
+                if (skullTrophy.inventory.itemList.Count == 1)
+                {
+                    var item = skullTrophy.inventory.itemList[0];
+                    item.RemoveFromContainer();
+                    item.Remove();
+                }
+
+                var skull = ItemManager.CreateByPartialName("skull.human");
+                skull.name = HumanBodyResourceDispenser.CreateSkullName(skullName);
+                skullTrophy.inventory.Insert(skull);
```

Comment:
I suggest using `skull.MoveToContainer(skullTrophy.inventory)`, and removing the item if that method returns `false`.